libcaf_core/src/uri.cpp

```diff
@@ -101,6 +102,27 @@ std::string to_string(const uri& x) {
   return result;
 }
 
+std::string to_string(const uri::authority_type& x) {
+  std::string str;
+  if (!x.userinfo.empty()) {
+    detail::append_uri(str, x.userinfo);
+    str += '@';
+  }
+  auto addr = get_if<ip_address>(&x.host);
+  if (addr == nullptr) {
+    detail::append_uri(str, get<std::string>(x.host));
+  } else {
+    str += '[';
```

Comment:
IPv4 addresses should not use the `[]` notation. It's "technically valid", because any IPv4 address also is an IPv6 address, but `to_string` output is meant for humans. 🙂

```c++
if (addr->embeds_v4()) // omit parentheses
else // print parentheses
```